### PR TITLE
[Search v2] [App] Create search results page

### PR DESCRIPTION
**Pull request by WojtekBoman**
### Details

This PR adjusts the header of the Search page to display the custom query. The header displays the user's query when the route param `isCustomQuery` is set to `true`. This PR covers also adding the `Filters` button to the header.
### Fixed Issues

$ https://github.com/Expensify/App/issues/45028
PROPOSAL:

### Tests

- [ ] Verify that no errors appear in the JS console

1. Navigate to the url with the `isCustomQuery` route param set to true, for example: https://dev.new.expensify.com:8082/search?q=type%3Aexpense%20status%3Aall&isCustomQuery=true
2. Verify if the header is displayed correctly

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/8b8ca54d-cb4e-4182-ac24-fa239e5a0b80

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/263e1041-6735-43a8-a65c-66863dcc4eb4

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/c47b46e5-e7ab-47a1-9e53-677f1f00b79c

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/bc9501c4-e10e-42ef-84d3-4783b2fa00e3

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/f6583d16-dd75-41df-b5e4-7be97c808d4e

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/70bc34c8-4382-4ca9-ab3e-3bb6454db198

</details>

### CODE REVIEW

File: src/components/Search/SearchListWithHeader.tsx [hunk 1]
```diff
@@ -44,7 +45,7 @@ function mapToItemWithSelectionInfo(item: TransactionListItemType | ReportListIt
 }
 
 function SearchListWithHeader(
-    {ListItem, onSelectRow, status, hash, data, searchType, isMobileSelectionModeActive, setIsMobileSelectionModeActive, ...props}: SearchListWithHeaderProps,
+    {ListItem, onSelectRow, queryJSON, hash, data, searchType, isMobileSelectionModeActive, setIsMobileSelectionModeActive, isSearchResultsMode = false, ...props}: SearchListWithHeaderProps,
```
Comment (Kicu) [inline src/components/Search/SearchListWithHeader.tsx:48]: ok hear me out 😅 controversial suggestion:
rename
```
isSearchResultsMode -> isCustomQueryMode`
```

Usually Im happy with names specific to what they describe and "mode" of SearchResults sounds like a good name.

However Search is becoming more and more complex. And in here we're introducing 2 new concepts: "SerchResultsMode" and "isCustomQuery".
So maybe everything will be simpler if we just drop the results naming, and use naming related to `customQuery` everywhere?

File: src/components/Search/SearchPageHeader.tsx [hunk 2]
```diff
@@ -12,39 +13,50 @@ import useTheme from '@hooks/useTheme';
...
     setIsMobileSelectionModeActive?: (isMobileSelectionModeActive: boolean) => void;
+    isSearchResultsMode?: boolean;
     setOfflineModalOpen?: () => void;
     setDownloadErrorModalOpen?: () => void;
 };
 
 type SearchHeaderOptionValue = DeepValueOf<typeof CONST.SEARCH.BULK_ACTION_TYPES> | undefined;
 
+const headerContent: {[key in SearchStatus]: {icon: IconAsset; titleTx: TranslationPaths}} = {
+    all: {icon: Illustrations.MoneyReceipts, titleTx: 'common.expenses'},
```
Comment (Kicu) [inline src/components/Search/SearchPageHeader.tsx:45]: I think this might be the right moment for us to switch to correct naming and use `common.all` translation in here.
What do you think @luacmartins ?

File: src/components/Search/SearchPageHeader.tsx [hunk 3]
```diff
@@ -56,13 +68,31 @@ function SearchPageHeader({
...
     const {setSelectedTransactionIDs} = useSearchContext();
+    const {status} = queryJSON;
+
+    const subtitle = useMemo(() => {
+        if (!isSearchResultsMode) {
+            return '';
+        }
+
+        return translate('search.filtersHeader');
+    }, [isSearchResultsMode, translate]);
```
Comment (Kicu) [inline src/components/Search/SearchPageHeader.tsx:79]: using of `useMemo` is a great practice, but to me in here just for 1 call to `translate` I find its not really needed

File: src/pages/Search/SearchPage.tsx [hunk 4]
```diff
@@ -17,7 +17,9 @@ type SearchPageProps = StackScreenProps<AuthScreensParamList, typeof SCREENS.SEA
 function SearchPage({route}: SearchPageProps) {
     const {shouldUseNarrowLayout} = useResponsiveLayout();
     const styles = useThemeStyles();
+    const {isCustomQuery} = route.params;
```
Comment (Kicu) [inline src/pages/Search/SearchPage.tsx:20]: can you please destructure all the params: `isCustomQuery`, `q` and `policyIDs` and use them in line 25?
that way all the params are in one place used the same way

File: src/pages/Search/SearchPageBottomTab.tsx [hunk 5]
```diff
@@ -34,6 +34,7 @@ function SearchPageBottomTab() {
         return {
             queryJSON: buildSearchQueryJSON(searchParams.q, searchParams.policyIDs),
             policyIDs: searchParams.policyIDs,
+            isSearchResultsMode: searchParams.isCustomQuery !== 'false',
```
Comment (Kicu) [inline src/pages/Search/SearchPageBottomTab.tsx:37]: NAB, not very important, but I'd move it out to a helper function even if it's a 1-liner.

**Review by Kicu — CHANGES_REQUESTED** [review]
Good job, code looks pretty nice I left some minor comments

File: src/pages/Search/SearchStatusMenuNarrow.tsx [hunk 6]
```diff
@@ -40,6 +41,10 @@ function SearchStatusMenuNarrow({statusMenuItems, activeItemIndex}: SearchStatus
         containerStyle: index === activeItemIndex ? [{backgroundColor: theme.border}] : undefined,
     }));
 
+    const menuIcon = useMemo(() => (title ? Expensicons.Filters : popoverMenuItems[activeItemIndex]?.icon ?? Expensicons.Receipt), [activeItemIndex, popoverMenuItems, title]);
+    const menuTitle = useMemo(() => title ?? popoverMenuItems[activeItemIndex]?.text, [activeItemIndex, popoverMenuItems, title]);
+    const titleViewStyles = title ? {...styles.flex1, ...styles.justifyContentCenter} : {};
```
[the remark this answers is not in the document]
Comment (WojtekBoman) [reply src/pages/Search/SearchStatusMenuNarrow.tsx:48]: This change throws a typescript error

File: src/components/Header.tsx [hunk 7]
```diff
@@ -21,9 +21,12 @@ type HeaderProps = {
 
     /** Additional header container styles */
     containerStyles?: StyleProp<ViewStyle>;
+
+    /** Whether the subtitle should be displayed above the title */
+    showSubtitleAboveTitle?: boolean;
 };
 
-function Header({title = '', subtitle = '', textStyles = [], containerStyles = [], shouldShowEnvironmentBadge = false}: HeaderProps) {
+function Header({title = '', subtitle = '', textStyles = [], containerStyles = [], shouldShowEnvironmentBadge = false, showSubtitleAboveTitle = false}: HeaderProps) {
```
Comment (luacmartins) [inline src/components/Header.tsx:29]: NAB supertitle is a common word used for this subtitle above title
```suggestion
function Header({title = '', subtitle = '', textStyles = [], containerStyles = [], shouldShowEnvironmentBadge = false, shouldShowSupertitle = false}: HeaderProps) {
```

File: src/components/Search/SearchListWithHeader.tsx [hunk 8]
```diff
@@ -14,15 +14,16 @@ import * as SearchUtils from '@libs/SearchUtils';
...
 
 type SearchListWithHeaderProps = Omit<BaseSelectionListProps<ReportListItemType | TransactionListItemType>, 'onSelectAll' | 'onCheckboxPress' | 'sections'> & {
-    status: SearchStatus;
+    queryJSON: SearchQueryJSON;
     hash: number;
     data: TransactionListItemType[] | ReportListItemType[];
     searchType: SearchDataTypes;
     isMobileSelectionModeActive?: boolean;
     setIsMobileSelectionModeActive?: (isMobileSelectionModeActive: boolean) => void;
+    isCustomQueryMode?: boolean;
```
Comment (luacmartins) [inline src/components/Search/SearchListWithHeader.tsx:26]:
```suggestion
    isCustomQuery?: boolean;
```

File: src/components/Search/SearchPageHeader.tsx (hunk 2 above)
Comment (luacmartins) [reply src/components/Search/SearchPageHeader.tsx:45]: I think we can keep `Expenses` until Search v2.2 when we introduce the other types and naming.

File: src/libs/Navigation/types.ts [hunk 9]
```diff
@@ -68,7 +68,7 @@ type CentralPaneScreensParamList = {
     // Param types of the search central pane are also used for the search bottom tab screen.
     [SCREENS.SEARCH.CENTRAL_PANE]: {
         q: SearchQueryString;
-        isCustomQuery: boolean;
+        isCustomQuery: string;
```
Comment (luacmartins) [inline src/libs/Navigation/types.ts:71]: Why this change? `isCustomQuery` implies a bool to me

File: src/pages/Search/SearchPage.tsx [hunk 10]
```diff
@@ -17,10 +18,11 @@ type SearchPageProps = StackScreenProps<AuthScreensParamList, typeof SCREENS.SEA
 function SearchPage({route}: SearchPageProps) {
     const {shouldUseNarrowLayout} = useResponsiveLayout();
     const styles = useThemeStyles();
+    const {policyIDs, q} = route.params;
 
-    const {policyIDs} = route.params;
+    const isCustomQueryMode = SearchUtils.isCustomQuery(route.params);
```
Comment (luacmartins) [inline src/pages/Search/SearchPage.tsx:23]: Let's rename all of these instances
```suggestion
    const isCustomQuery = SearchUtils.isCustomQuery(route.params);
```

File: src/pages/Search/SearchStatusMenuNarrow.tsx [hunk 11]
```diff
@@ -65,6 +78,11 @@ function SearchStatusMenuNarrow({statusMenuItems, activeItemIndex}: SearchStatus
                     </Animated.View>
                 )}
             </PressableWithFeedback>
+            <Button
+                icon={Expensicons.Filters}
+                onPress={() => Navigation.navigate(ROUTES.SEARCH_ADVANCED_FILTERS)}
+                innerStyles={styles.searchNarrowFilterButton}
+            />
```
Comment (luacmartins) [inline src/pages/Search/SearchStatusMenuNarrow.tsx:85]: Same here
```suggestion
```

**Review by luacmartins — CHANGES_REQUESTED** [review]
Looking good. Left a few comments

BUG: Other filters don't show up in the supertitle. For example, manually changing the URL to `/search?q=type%3Aexpense%20status%3Aall%20category%3AUncategorized&isCustomQuery=true` still shows the canned search, instead of `type:expense status:all category:Uncategorized`

Screenshot 2024-07-26 at 9 23 50 AM

**Comment by ikevin127** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by ikevin127** [comment]
@WojtekBoman 👋 Hi, I noticed some discrepancies while reviewing the PR ⬇️ 

**Tests state:**
> 1. Navigate to the url with the isCustomQuery route param set to true, for example: https://dev.new.expensify.com:8082/search?q=type%3Aexpense%20status%3Aall&isCustomQuery=true
> 2. Verify if the header is displayed correctly

While (1.) works as expected functionaloty-wise, when it comes to (2.) -> I found some UI related discrepancies (cc @Expensify/design) when compared to the [design doc](https://docs.google.com/document/d/1bRV1nPz8SaI_6DHqfehmLnADxv40us_q0JNvmZ1QyaY/edit#heading=h.n54f9gkc6gf6) which I think are within scope for this PR (correct me if I'm wrong):

* [ ] **1. Wide Layout - Filter labels are lowercase and missing spaces**

    **Note:** This can only be seen on wide layout (Web / Desktop).
    | Design Doc | PR | PR - Narrow layout (correct) |
    | ------------- | ------------- | ------------- |
    | design doc wide | Screenshot 2024-07-26 at 14 39 06 | Screenshot 2024-07-26 at 15 04 25 |

* [ ] **2. Narrow Layout - Incorrect filter dropdown icon color and padding between icons and text**

    **Note:** This can only be seen on narrow layout (mWeb / Native).
    | Design Doc | PR | ------ Current Staging ------ (incorrect as well) |
    | ------------- | ------------- | ------------- |
    | design doc narrow | Screenshot 2024-07-26 at 14 43 10 | Screenshot 2024-07-26 at 14 07 07 |

* [ ] **3. Narrow Layout - Filter button icon not centered on native app**

    **Note:** This can only be seen on narrow layout and only on iOS & Android: Native. Click on pictures below to get zoomed-in version.
    | Design Doc | PR - iOS | PR - Android |
    | ------------- | ------------- | ------------- |
    | design doc narrow | Screenshot 2024-07-26 at 14 49 00 | Screenshot 2024-07-26 at 14 49 36 |

* [x] **4. Narrow Layout - Filter dropdown opens search selection modal** (this one is probably NAB but I wanted to bring this up because I wasn't sure)

    **Note:** This can only be seen on narrow layout.
    <details>
    <summary>Video</summary>
    
    https://github.com/user-attachments/assets/cda3ffab-4aeb-45dd-bdd7-1c293d4a981d

    </details>

    Is ^ this supposed to work this way ? Asking since it makes sense to have the small Filters button on the right side which opens the Filters page, but clicking the wider filter select dropdown button -> I'd expect to open the Filters page and not the Search page selection modal as this select dropdown button normally does:
    
    Screenshot 2024-07-26 at 15 02 08

##

♻️ Once the above discrepancies are addressed / resolved, I will be able to complete the reviewer checklist and **Approve**.

cc @luacmartins

File: src/libs/Navigation/types.ts (hunk 9 above)
Comment (WojtekBoman) [reply src/libs/Navigation/types.ts:71]: It's a bit misleading because this parameter is stored as a string in `route.params`, and we need to parse it to `boolean` when we read it in the component

**Comment by WojtekBoman** [comment]
@ikevin127 

Regarding the first point, I relied on these designs from figma: https://www.figma.com/design/YHaWaw3F0PgxqI0tXHJVeY/Search-v1-v3?node-id=3942-24214&t=192BImUkKvmas9Ks-0

Screenshot 2024-07-29 at 08 41 57

cc: @shawnborton 

I'll take a look at the rest of your comments 🫡

**Comment by shawnborton** [comment]
But taking these in turn...

> Filter labels are lowercase and missing spaces

What do you think @luacmartins?

> Incorrect filter dropdown icon color and padding between icons and text

Given that the app currently does not use a green icon for this, maybe we can ignore this for now:
CleanShot 2024-07-29 at 09 17 34@2x


> Narrow Layout - Filter button icon not centered on native app

Yeah, we should have a specific component that is for buttons that do not have any labels and just have an icon. Let's make sure we're using that here. 

> 4. Narrow Layout - Filter dropdown opens search selection modal (this one is probably NAB but I wanted to bring this up because I wasn't sure)

That looks correct to me.

File: src/libs/Navigation/types.ts (hunk 9 above)
Comment (Kicu) [reply src/libs/Navigation/types.ts:71]: yeah this looks surprising but I believe its correct to type it like this. Everything that comes from route query params will always be a string I believe

**Comment by dubielzyk-expensify** [comment]
> highlight the current page that is selected in the bottom sheet

What would the "current page" mean in that instance, @shawnborton ?

Also, I'm not sure if it's been reported, but the bottom sheet is missing the scrim/overlay:

CleanShot 2024-07-29 at 19 42 22@2x

Here's a partial mock with scrim/overlay:

CleanShot 2024-07-29 at 19 43 26@2x

**Comment by shawnborton** [comment]
> What would the "current page" mean in that instance, @shawnborton ?

Basically this mock here where we add the current filtered view as an item in the Search navigation bottom sheet:
CleanShot 2024-07-29 at 12 11 59@2x

**Comment by WojtekBoman** [comment]
> > What would the "current page" mean in that instance, @shawnborton ?
> 
> Basically this mock here where we add the current filtered view as an item in the Search navigation bottom sheet: CleanShot 2024-07-29 at 12 11 59@2x

I understand that this button should close the modal when we tap it, right?

**Comment by shawnborton** [comment]
I'm not quite following - do you mind rephrasing?

**Comment by shawnborton** [comment]
Ah okay cool, thanks for clarifying and yes, what you are showing above looks correct to me. It would just close out the modal.

**Comment by shawnborton** [comment]
Also one thing I noticed... it looks like the button isn't wide enough to match the content below it, I bet that's just some leftover margin you were using when the filter button is present:
CleanShot 2024-07-29 at 15 59 29@2x

**Comment by WojtekBoman** [comment]
> Also one thing I noticed... it looks like the button isn't wide enough to match the content below it, I bet that's just some leftover margin you were using when the filter button is present: CleanShot 2024-07-29 at 15 59 29@2x

Good catch! Thanks for noticing that, I'll fix that :)

**Comment by WojtekBoman** [comment]
> Looking good. Left a few comments
> 
> BUG: Other filters don't show up in the supertitle. For example, manually changing the URL to `/search?q=type%3Aexpense%20status%3Aall%20category%3AUncategorized&isCustomQuery=true` still shows the canned search, instead of `type:expense status:all category:Uncategorized`
> 
> Screenshot 2024-07-26 at 9 23 50 AM

@luacmartins I've solved this issue, but another one appeared. For now I use regex to remove sortOrder and sortBy from the input, but with that solution the header text is lowercase, I could capitalize this text by splitting the text by an empty space but it will lead to issues when we have a two words value of the param. To me, we should loop through the AST tree and parse the values ​​stored there to the capitalized text. I'll try to take a look at it tomorrow, if it's urgent to merge this PR, can we create a follow-up?

**Comment by luacmartins** [comment]
>>Filter labels are lowercase and missing spaces

>What do you think @luacmartins?

IMO we should leave the query unaltered, i.e. with lowercase keys and no commas between them, e.g. `type:expense status:all category:travel`. This is what users will be typing into the search bar input later on, so changing this input might educate them to use the wrong syntax later on.

**Comment by luacmartins** [comment]
>To me, we should loop through the AST tree and parse the values ​​stored there to the capitalized text.

We should use [SearchUtils.getFilters](https://github.com/Expensify/App/blob/410a6f21524ddc1a4f85fbf9f922c6bc21d7e502/src/libs/SearchUtils.ts#L403) and pick only the filters that are needed, i.e. everything in `SYNTAX_FILTER_KEYS` and `type` and `status` in `SYNTAX_ROOT_KEYS` https://github.com/Expensify/App/blob/410a6f21524ddc1a4f85fbf9f922c6bc21d7e502/src/CONST.ts#L5289-L5310

File: src/libs/SearchUtils.ts [hunk 12]
```diff
@@ -411,20 +412,31 @@ function getFilters(query: SearchQueryString, fields: Array<Partial<AllFieldKeys
     return filters;
 }
 
+function getSearchHeaderTitle(input: string) {
```
Comment (luacmartins) [inline src/libs/SearchUtils.ts:415]: Left a comment about this, but I think we can use `SearchUtils.getFilters`

**Comment by ikevin127** [comment]
> Also, I'm not sure if it's been reported, but the bottom sheet is missing the scrim/overlay:

@dubielzyk-expensify That's only happening on local dev due to `StrictMode`, I debunked it in https://github.com/Expensify/App/pull/46096#issuecomment-2252097059.

**Comment by WojtekBoman** [comment]
I've prepared a more complex query to test how it would work:
`type:expense status:all category:Uncategorized,Categorized category!=wrongCategory,wrongCategory2 amount>200 amount<1000 date>02-05-2020 date<04-05-2022`
 @shawnborton Could you take a look if it looks good or if something needs to be improved?
Screenshot 2024-07-30 at 12 57 22

**Comment by WojtekBoman** [comment]
@ikevin127 All comments have been resolved :) Could you recheck it?

**Review by luacmartins — APPROVED** [review]
LGTM and tests well. @ikevin127 all yours!

**Review by ikevin127 — APPROVED** [review]
LGTM 🚀

**Review by luacmartins — DISMISSED** [review]

**Comment by luacmartins** [comment]
@WojtekBoman we have conflicts 😞 Otherwise, looks good to be merged

**Comment by WojtekBoman** [comment]
@luacmartins Solved :)

**Review by luacmartins — APPROVED** [review]

**Comment by luacmartins** [comment]
@WojtekBoman :hero:
